docs/source/workflow/generate-types.mdx

```diff
@@ -0,0 +1,219 @@
+
+```yaml
+# This configuration file tells GraphQL Code Generator how
+# to generate types based on our schema.
+schema: "./schema.graphql"
+generates:
+  # Specify where our generated types should live.
+  ./src/__generated__/resolvers-types.ts:
+    plugins:
+      - "typescript"
+      - "typescript-resolvers"
+    config:
+      useIndexSignature: true
+      # More on this below!
+      contextType: ../index#MyContext
+```
+
+> [See the docs](https://www.the-guild.dev/graphql/codegen/plugins/typescript-resolvers#usage-example) for more information on the above configuration options.
+
+Finally, we recommend adding helpful scripts to your `package.json` file to ensure your TS types are regularly generated:
+
+```json
+{
+// ...
+  "scripts": {
+    "compile": "graphql-codegen --config codegen.yml && tsc",
+    "postinstall": "npm run compile",
+    "start": "npm run compile && node ./dist/index.js",
+  },
+// ...
+}
+```
+
+Above, running the `npm start` command generates types based on our GraphQL schema _and_ compiles our TypeScript code. The first time you run the `graphql-codegen` command, you'll see a file full of generated types at the path you specified in your `codegen.yml` file. 
+
+## Adding types to resolvers
+
+The `typescript-resolvers` plugin creates a `Resolvers` type that you can use to add a type to your [resolver map](../data/resolvers#base-syntax), ensuring your resolvers return values match the field types specified by your schema.
+
+Import the `Resolvers` type into the file where you define your resolvers:
+
+```ts title="resolvers.ts"
+// This is the file where our generated types live
+// (specified in our `codegen.yml` file)
+import { Resolvers } from './__generated__/resolvers-types';
+```
+
+You can now add the `Resolvers` type directly to your resolver map:
+
+```ts
+export const resolvers: Resolvers = {}
+```
+
+Your resolvers can now type check that the *arguments* and *return value* for each resolver match the schema:
+
+```ts
+export const resolvers: Resolvers = {
+  Query: {
+    // TypeScript now complains about the below resolver because
+    // the data returned by this resolver doesn't match the schema type
+    // (i.e., type Query { books: [Book] })
+    books: () => {
+      return "apple";
+    },
+  },
+}
+```
+
+### Context typing for resolvers
+
+You can also configure GraphQL Code Generator to add a type for the context your resolvers share, ensuring TypeScript warns you if you attempt to use a context value that doesn't exist.
+
+To do this, you must first export the interface you pass to Apollo Server as a generic type parameter for typing your context value:
+
+```ts title="src/index.ts"
+// highlight-start
+export interface MyContext {
+  dataSources: {
+    books: Book[];
+  };
+}
+// highlight-end
+
+const server = new ApolloServer<MyContext>({
+  typeDefs,
+  resolvers,
+});
+```
+
+Next, add a `contextType` configuration option to your `codegen.yml`:
```

Comment:
We did already do this in the original config above with a "more on this later" note. Maybe we can get away with not reprinting the whole yaml here again and just allude back to the note "Remember the `contextType` we added in our config earlier? ..." or something like that.

Reply:
I like it!